HeterogeneousCore/CUDAUtilities/interface/getCudaDrvErrorString.h

```diff
@@ -0,0 +1,17 @@
+#ifndef HeterogeneousCore_CUDAUtilities_getCudaDrvErrorString_h
+#define HeterogeneousCore_CUDAUtilities_getCudaDrvErrorString_h
+
+#include "cuda.h"
+
+inline const char *getCudaDrvErrorString(CUresult error_id) {
+  const char *message;
+  auto ret = cuGetErrorName(error_id, &message);
```

Comment:
This function was originally added in a6bc36ae6585fa9a14a2ee40862b3050b282795d with a "copy-pasted" array of error codes and their string counterparts, and then moved around in fa62ef125f57d2f3596c65409037932546d82a40. But while preparing this PR I noticed this
https://github.com/cms-patatrack/cmssw/blob/c0cd5117e0e553a4416324a73f5ac05f0621e901/HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h#L14-L17
and realized that very likely the "copy-pasted" array can be replaced with a call to `cuGetErrorName()`. Nevertheless I think the function has some added value on top of `cuGetErrorName()` by returning the C-string.

What do others think?

Also is the current level of cleanup (in commits 88404a1f1db2cc26d7178346740419aac57b587d and c28a0848dea98263185f1ce7123fd2fd22d638e2) good-enough or would people prefer squashing them around (so that the "copy-pasted" array or `BuildFile.xml` edits do not show up in history)?

Reply:
I don't mind having the extra function around - if we end up not using it, we can drop it later on.

As for the commits history: I've been squashing most of the PR because they tended to have a long history of not-very-meaningful commits; IMHO the commit history in the PR should reflect the changes we want to make, not how we got there.

For this PR, I would keep more than one commit, but I would
  - squash 08c0dbf and ff8a8bf into f12bdf4
  - squash c28a084, 88404a1 and fa62ef1 into a6bc36a

Something like
```
pick    f12bdf4 Add an unit test for CUDAService.
squash  08c0dbf Improved CUDAService unit test: if the runtime API is working, the service is guaranteed to be enabled, and it runs the checks on the getters.
squash  ff8a8bf Minor fixes for testCUDAService unit test.

pick    06ff473 Added new CUDAService unit test getting information from the driver API.

pick    a6bc36a Improve Error handling in CUDAService.
squash  fa62ef1 Factor out CUDA driver API error string utility code into a standalone header on a dedicated package, and include it in CUDAService.
squash  88404a1 Use cuGetErrorName() instead of duplicating the error names
squash  c28a084 Remove unnecessary additions to BuildFiles

reword  a62a0d6 Do not dlopen libcuda.so: this is installation and/or ld.conf dependent, so just rely on the -l cuda in the linker, as set up by cuda-stubs external.

reword  a6ab6d9 Bug fix: if the CUDAService is disabled by configuration, the ctor just returns, so it does not set any data member. This has been found in the unit test by explicitly forcing the configuration in the CUDAService PSet to be disabled: the last assert fails.
```

Reply:
@makortel how would you like me to proceed ?
  - clean up the history, push the changes to your branch, and merge it (PR will show up as "merged")
  - same but leave your two commits un-squashed
  - clean up the commits locally and merge it (PR will show up as "closed")
  - same but leave your two commits un-squashed
  - merge the PR, then clean up the history (PR should still show up as "merged")
  - same but leave your two commits un-squashed

Reply:
@fwyzard I'll update the branch later today (sorry for delay...).